NuRadioReco/modules/voltageToEfieldConverter.py

```diff
@@ -188,27 +188,11 @@ def run(self, evt, station, det, use_channels=None, use_MC_direction=False, forc
         else:
             efield3_f[1:, mask] = np.moveaxis(stacked_lstsq(np.moveaxis(efield_antenna_factor[:, :, mask], 2, 0), np.moveaxis(V[:, mask], 1, 0)), 0, 1)
 
-        efield_position = np.mean([
-            det.get_relative_position(station.get_id(), channel_id)
-            for channel_id in use_channels], axis=0)
-
         electric_field = NuRadioReco.framework.electric_field.ElectricField(use_channels, efield_position)
         electric_field.set_frequency_spectrum(efield3_f, station.get_channel(use_channels[0]).get_sampling_rate())
         electric_field.set_parameter(efp.zenith, zenith)
         electric_field.set_parameter(efp.azimuth, azimuth)
-        # figure out the timing of the E-field
-        t_shifts = np.zeros(V.shape[0])
-        site = det.get_site(station_id)
-        if(zenith > 0.5 * np.pi):
-            logger.warning("Module has not been optimized for neutrino reconstruction yet. Results may be nonsense.")
-            refractive_index = ice.get_refractive_index(-1, site)  # if signal comes from below, use refractivity at antenna position
-        else:
-            refractive_index = ice.get_refractive_index(1, site)  # if signal comes from above, in-air propagation speed
-        for i_ch, channel_id in enumerate(use_channels):
-            antenna_position = det.get_relative_position(station.get_id(), channel_id)
-            t_shifts[i_ch] = station.get_channel(channel_id).get_trace_start_time() - geo_utl.get_time_delay_from_direction(zenith, azimuth, antenna_position, n=refractive_index)
-
-        electric_field.set_trace_start_time(t_shifts.max())
+        electric_field.set_trace_start_time(times[0])
```

Comment:
may also need to be revisited depending on how you solve the 'large `t_shifts` issue mentioned above, I think.

Reply:
This is fine, no?